### PR TITLE
Add status check enforcement for column_family_test.cc

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: db/column_family_test.cc [hunk 1]
```diff
@@ -280,7 +282,8 @@ class ColumnFamilyTestBase : public testing::Test {
       ColumnFamilyDescriptor desc;
       ASSERT_OK(handles_[cfi]->GetDescriptor(&desc));
       RocksDBOptionsParser::VerifyCFOptions(ConfigOptions(), desc.options,
-                                            current_cf_opt);
+                                            current_cf_opt)
+          .PermitUncheckedError();
```
Comment (mrambacher) [inline db/column_family_test.cc:286]: Why is this status being ignored?  After spending a lot of time in the Options code, I am especially curious about this one...

**Review by mrambacher — APPROVED** [review]
LGTM.

There are several other PRs that are also changing db_impl_write at the same time.  I have suggested a new signature to the "BGError" so that the status does not need to be ignored everywhere.

File: db/column_family_test.cc [hunk 2]
```diff
@@ -77,11 +77,13 @@ class ColumnFamilyTestBase : public testing::Test {
 
   ~ColumnFamilyTestBase() override {
     std::vector<ColumnFamilyDescriptor> column_families;
+#ifndef ROCKSDB_LITE
```
Comment (riversand963) [inline db/column_family_test.cc:80]: Maybe always call `GetDescriptor`, and check the return value separately.

File: db/column_family_test.cc (hunk 1 above)
Comment (akankshamahajan15) [reply db/column_family_test.cc:286]: In the test itself it expects the arguments to be invalid so it returns invalid in few cases and OK in other.

**Comment by akankshamahajan15** [comment]
> LGTM.
> 
> There are several other PRs that are also changing db_impl_write at the same time. I have suggested a new signature to the "BGError" so that the status does not need to be ignored everywhere.

Sounds good. I will add the comment as TODO and when its confirmed to add a new signature to return void, I will change the signature here as well.

File: db/column_family_test.cc (hunk 1 above)
Comment (akankshamahajan15) [reply db/column_family_test.cc:286]: @mrambacher, Oh my bad. This is different. It gives an error that 
"RocksDBOptionsParser::VerifyCFOptions( ConfigOptions(), desc.options, current_cf_opt)
Invalid argument: [RocksDBOptionsParser]: failed the verification on ColumnFamilyOptions::periodic_compaction_seconds--- The specified one is 0 while the persisted one is 18446744073709551614."
Line: https://github.com/facebook/rocksdb/blob/7d472accdca996d7d83ae8ce78ad17f799696926/options/options_parser.cc#L679

Comment (akankshamahajan15) [reply db/column_family_test.cc:286]: I am not sure if this is expected or not.

Comment (mrambacher) [reply db/column_family_test.cc:286]: I do not think that is expected.  I will take a quick look...

Comment (mrambacher) [reply db/column_family_test.cc:286]: The problem is that when you add a ColumnFamily to the DB, the ColumnFamilyOptions get sanitized and some of the default values might change.  If you change the line to this:
      // Need to sanitize the default column family options before comparing them...
      ASSERT_OK(RocksDBOptionsParser::VerifyCFOptions(ConfigOptions(), desc.options,
                                                      SanitizeOptions(dbfull()->immutable_db_options(), current_cf_opt)));

The tests should pass.

Comment (akankshamahajan15) [reply db/column_family_test.cc:286]: Yes that works. Thanks a lot!

**Review by riversand963 — APPROVED** [review]

**Pull request closed without merging**